google/cloud/sql/connector/instance.py

```diff
@@ -74,6 +75,15 @@ def __init__(
         self._refresh_in_progress = asyncio.locks.Event()
         self._current: asyncio.Task = self._schedule_refresh(0)
         self._next: asyncio.Task = self._current
+        self._closed = False
+
+    @property
+    def conn_name(self) -> ConnectionName:
+        return self._conn_name
+
+    @property
+    def closed(self) -> bool:
+        return self._closed
 
     async def force_refresh(self) -> None:
```

Comment:
This should probably balk if the instance is closed: 
```python
async def force_refresh(self) -> None:
    if self._closed: 
        return
```
Same for `_perform_refresh(self)` and `connect_info(self)`

Reply:
I put these on the MonitoredCache class to not have to do on both background and lazy caches.